CMakeLists.txt

```diff
@@ -228,6 +228,24 @@ endif()
 # GNUInstallDirs.
 include(GNUInstallDirs)
 
+find_package(nlohmann_json QUIET)
+if(NOT nlohmann_json_FOUND)
+  message("Using local nlohmann::json from submodule")
+  set(JSON_BuildTests
+      OFF
+      CACHE INTERNAL "")
+  set(JSON_Install
+      OFF
+      CACHE INTERNAL "")
+  # This option allows to link nlohmann_json::nlohmann_json target
+  add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/nlohmann-json)
```

Comment:
Will it fail `cmake` configure if submodules are not checked out? Just thinking of the scenarios if the user wants to build api and sdk which ideally doesn't have json/curl requirements, we shouldn't let configure fail in those cases.

Reply:
Since it is required for _most_ build configurations, I can add a check if directory exists. If it does not exist, I can trigger a warning recommending to do recursive clone.

Reply:
Yes agreed. How about not breaking the build if `WITH_API` is enabled and `json` is not installed? Though I don't have strong opinion on this :)

Reply:
@lalitb - Sounds good. I can improve it in one of my other build infra PRs!